packages/eslint-plugin/lib/rules/require-img-alt.js

```diff
@@ -77,16 +142,37 @@ module.exports = {
 /**
  * @param {Tag} node
  * @param {string[]} substitute
- * @returns
+ * @returns {{hasAnyAlt: boolean, hasEmptyAlt: boolean, hasValidContent: boolean}}
  */
-function hasAltAttrAndValue(node, substitute = []) {
-  return node.attributes.some((attr) => {
-    if (attr.key && attr.value) {
-      return (
-        (attr.key.value === "alt" || substitute.includes(attr.key.value)) &&
-        typeof attr.value.value === "string"
-      );
+function hasValidAltOrSubstitute(node, substitute = []) {
```

Comment:
```suggestion
function hasValidAltOrSubstitute(node, substitute) {
```
IMO, we don't need to put a default value.